OSPSuite.Presentation.Importer/Core/DataFormat/ColumnsDataFormat.cs

```diff
@@ -0,0 +1,150 @@
+﻿using System.Text.RegularExpressions;
+using System.Collections.Generic;
+using System.Linq;
+using System;
+
+namespace OSPSuite.Presentation.Importer.Core.DataFormat
+{
+   public class ColumnsDataFormat : IDataFormat
+   {
+      public string Name => "ColumnsFormat";
+
+      public string Description => "Simple format with data identified by column data...";
+
+      public IList<DataFormatParameter> Parameters { get; private set; }
+
+      public bool CheckFile(IUnformattedData data)
+      {
+         if (data.Headers.Where(h => h.Value.Level == ColumnDescription.MeasurementLevel.Numeric).Count() < 2)
+            return false;
+         setParameters(data);
+         return true;
+      }
+
+      private void setParameters(IUnformattedData data)
+      {
+         var keys = data.Headers.Keys.ToList();
+         Parameters = new List<DataFormatParameter>();
+
+         var missingKeys = new List<string>();
+
+         extractQualifiedHeadings(keys, missingKeys);
+         extractNonQualifiedHeadings(keys, missingKeys, data);
+         extractGeneralParameters(keys, data);
+      }
+
+      private string extractUnits(string description)
+      {
+         var units = Regex.Match(description, @"\[.+\]").Value;
+         return units.Substring(1, units.Length - 2).Trim();
+      }
+
+      private void extractQualifiedHeadings(List<string> keys, List<string> missingKeys)
+      {
+         foreach (var header in Enum.GetNames(typeof(Column.ColumnNames)))
+         {
+            var headerKey = keys.FirstOrDefault(h => h.ToUpper().Contains(header.ToUpper()));
+            if (headerKey != null)
+            {
+               keys.Remove(headerKey);
+               Column.ColumnNames columnName;
+               Enum.TryParse(header, out columnName);
+               Parameters.Add(new MappingDataFormatParameter(headerKey, new Column() { Name = columnName, Unit = extractUnits(headerKey) }));
+            }
+            else
+            {
+               missingKeys.Add(header);
+            }
+         }
+      }
+
+      private void extractNonQualifiedHeadings(List<string> keys, List<string> missingKeys, IUnformattedData data)
+      {
+         foreach (var header in missingKeys)
+         {
+            var headerKey = keys.FirstOrDefault
+               (h => 
+                  data.Headers[h].Level == ColumnDescription.MeasurementLevel.Numeric && 
```

Comment:
This is super hard to read. Let's simplify this. Muliple Where and select ... I am too dumb for this

Reply:
I hope it makes more sense now?